src/macro.ts

```diff
@@ -0,0 +1,126 @@
+// const printAST = require('ast-pretty-print')
+import type babelCore from '@babel/core'
+import {createMacro} from 'babel-plugin-macros'
+import type {MacroHandler} from 'babel-plugin-macros'
+import {getReplacement} from './helpers'
+
+const prevalMacros: MacroHandler = function prevalMacros({
+  references,
+  state,
+  babel,
+}) {
+  const fileOpts = state.file.opts
+  references.default.forEach(referencePath => {
+    const path = referencePath as babelCore.NodePath<babelCore.types.Identifier>
+    const targetPath = path.parentPath
+    if (targetPath.type === 'TaggedTemplateExpression') {
+      asTag(
+        targetPath as babelCore.NodePath<babelCore.types.TaggedTemplateExpression>,
+        fileOpts,
+        babel,
+      )
+    } else if (targetPath.type === 'CallExpression') {
+      asFunction(
+        targetPath as babelCore.NodePath<babelCore.types.CallExpression>,
+        fileOpts,
+        babel,
+      )
+    } else if (targetPath.type === 'JSXOpeningElement') {
+      const jsxElement = targetPath.parentPath
+      asJSX(
+        jsxElement as babelCore.NodePath<babelCore.types.JSXOpeningElement>,
```

Comment:
These are somewhat copied from `babel-plugin-codegen` but I started wondering if `JSXOpeningElement` is the right type here. Should it be `JSXElement`? 

Inside `asJSX` the `get('children')` call would also return an array of JSX* types instead of `babelCore.NodePath<babelCore.types.Node> | babelCore.NodePath<babelCore.types.Node>[]`

Reply:
Thanks for taking notice @juhanakristian. Let's handle that in a new issue. I'd like to avoid changing too many things at once.